packages/core/src/install/index.ts

```diff
@@ -326,6 +327,14 @@ export async function mutateModules (
             },
             patchedDependencies: patchedDependenciesWithResolvedPath,
             projects: ctx.projects as Project[],
+            allProjects: opts.allProjects
+              ? (
+                await readProjectsContext(opts.allProjects.map(p => ({ ...p, rootDir: p.dir })), {
+                  lockfileDir: opts.lockfileDir,
+                  modulesDir: opts.modulesDir,
+                })
+              ).projects as unknown as Project[]
```

Comment:
one thing I haven't been able to resolve is here: the `projects` passed at `allProjects` are missing `buildIndex` field

Reply:
ok for this, buildIndex is now assigned in headless when it's needed